staging/src/k8s.io/apimachinery/pkg/apis/meta/internalversion/types.go

```diff
@@ -56,6 +56,10 @@ type ListOptions struct {
 	// it does not recognize and will return a 410 error if the token can no longer be used because
 	// it has expired.
 	Continue string
+	// A subset of fields of interest.  A watch will only send updates if one of these fields has changed.
+	// TODO: Name is not ideal... maybe `fieldMask` or `watchMask`.
+	// Also, do we want a separate concept for the fields we want to fetch, vs the fields we want to trigger notifications.
+	FieldSubset []string
```

Comment:
vote for watchMask :)

Reply:
But if we decide to go further, we may need to figure out how to identify fields in different depth, e.g. `A`, `spec.A`, `status.A`, `spec.xxx.A` etc.

Reply:
Ended up changing to fieldMask / watchMask :-)  I think we don't need to _implement_ it right away, and we don't necessarily need full field support, but I agree we should think about the syntax.  fieldMask actually is a Protobuf thing and has a partial specification https://developers.google.com/protocol-buffers/docs/reference/java/com/google/protobuf/FieldMask

Arrays (and maps) are the challenge - protobuf fieldmask just disallows them except at the leaf, which is probably good enough for us as well - and easily extended in future should we want to support something more like jsonpath / jmespath.